app/scripts/metamask-controller.js

```diff
@@ -4885,6 +4885,48 @@ export default class MetamaskController extends EventEmitter {
     } ${hdPathDescription || ''}`.trim();
   }
 
+  /**
+   * Get next available label for an hardware account
+   *
+   * @param {number} providedIndex - Index of the account in the accounts list
+   * @param {string} hardwareDeviceName - Name of the hardware device (Onekey, Ledger, etc)
+   * @param {string} hdPathDescription - HD path description
+   * @returns string label
+   */
+  getNextAvailableLabel(providedIndex, hardwareDeviceName, hdPathDescription) {
+    const label = (index) =>
+      this.getAccountLabel(hardwareDeviceName, index, hdPathDescription);
+    const existingLabels = this.accountsController
+      .listAccounts()
+      .map((account) => account.metadata.name);
+    let newIndex = providedIndex;
+    while (existingLabels.includes(label(newIndex))) {
+      newIndex += 1;
+    }
+    return this.getAccountLabel(
+      hardwareDeviceName,
+      newIndex,
+      hdPathDescription,
+    );
```

Comment:
Nit: when a valid label is found (by calling `label(newIndex)` -> `getAccountLabel`), `getAccountLabel` is called _one more time_ with the same exact parameters in the `return` statement. Technically better to cache and reuse the value.

Very Minor Nit: Also, if you convert `existingLabels` to a `Set`, the check done in the `while` loop should be much faster. `Set#has` is faster than `Array#includes` in worst case scenarios. For "small" datasets the extra overhead of creating the `Set` might make the code slower, though in the technical worst case (where there are 100s of accounts and the label name+index is already used) the Set would be faster.

Either nit above is fine to ignore, just notice it and thought I'd mention it.